Makefile

```diff
@@ -4,29 +4,31 @@ GOBIN = go
 # i.e. ignore local $GOPATH/src installed sources
 GOPATH_TMP = $(CURDIR)/.GOPATH
 GO = /usr/bin/env GOPATH=$(GOPATH_TMP) $(GOBIN)
-GO_FLAGS =
 GOFMT = gofmt
 VERSION = dev-$(shell date +%FT%T%z)
 
-OS = linux
-ARCH = amd64
+GO_OS = GOOS="${GOOS}"
+GO_ARCH = GOARCH="${GOARCH}"
 BINARY = kubeapps
 GO_PACKAGES = $(IMPORT_PATH)/cmd $(IMPORT_PATH)/pkg/...
 GO_FILES := $(shell find $(shell $(GOBIN) list -f '{{.Dir}}' $(GO_PACKAGES)) -name \*.go)
-GO_FLAGS = -ldflags="-w -X github.com/kubeapps/kubeapps/cmd.VERSION=${VERSION}"
+GO_FLAGS = -ldflags='-extldflags "-static" -w -X github.com/kubeapps/kubeapps/cmd.VERSION=${VERSION}'
+GO_LDFLAGS =
 EMBEDDED_STATIC = generated/statik/statik.go
+CC_ = CC="${CC}"
+CXX_ = CXX="${CXX}"
 
 default: binary
 
 binary: build-prep $(EMBEDDED_STATIC)
-	$(GO) build -i -o $(BINARY) $(GO_FLAGS) $(IMPORT_PATH)
+	CGO_ENABLED=1 $(CC_) $(CXX_) $(GO_OS) $(GO_ARCH) $(GO) build -i -o $(BINARY) $(GO_FLAGS) $(IMPORT_PATH)
 
 test: build-prep $(EMBEDDED_STATIC)
 	$(GO) test $(GO_FLAGS) $(GO_PACKAGES)
 
 $(EMBEDDED_STATIC): build-prep $(wilcard static/*)
-	$(GO) build -o statik ./vendor/github.com/rakyll/statik/statik.go
-	$(GO) generate
+	GOOS=linux $(GO) build -o statik ./vendor/github.com/rakyll/statik/statik.go
+	GOOS=linux $(GO) generate
```

Comment:
I don't understand what this does.  We force the statik executable to be built for linux .. and then run it even if the local arch isn't linux?

Reply:
I enforced the linux GOOS here because `go generate` can't run with GOOS="windows" - it executes the `statik` binary which leads to `cannot execute binary file: Exec format error`. But you're right: enforcing it to linux can't also work on osx. Hm... I am asking @sameersbn to help me run a normal build on windows machine to double check if `go generate` works on windows or not. Then I will figure out the proper setup for that line. Thanks Gus for this catch 👍

Reply:
`go generate` does not need to work on Windows... we run the `statik` binary on Linux which generates files that we then cross-compile for Windows. Presumably, the issue is that the OS X travis job follows this same pipeline and fails because we are forcing the build to Linux? Surely we just need to remove/unset `GOOS` for these two steps so it uses whatever OS it currently is running in?

Reply:
@prydonius 

> I enforced the linux GOOS here because go generate can't run with GOOS="windows" - it executes the statik binary which leads to cannot execute binary file: Exec format error.

If we don't put GOOS=linux on travis build for that line, it will fail at the GOOS=windows build job. I added a commit below to fix that case. It works just fine now.

Reply:
Okay, now I see that you're referring to when you're trying to build on a local machine. I believe setting `GOOS= $(GO) generate` should work.